src/prefect/agent/ecs/agent.py

```diff
@@ -224,14 +232,20 @@ def __init__(  # type: ignore
         else:
             self.run_task_kwargs = {}
 
-        # If running on fargate, auto-configure `networkConfiguration` for the
-        # user if they didn't configure it themselves.
-        if self.launch_type == "FARGATE" and not self.run_task_kwargs.get(
-            "networkConfiguration"
-        ):
-            self.run_task_kwargs[
-                "networkConfiguration"
-            ] = self.infer_network_configuration()
+        if not self.run_task_kwargs.get("capacityProviderStrategy"):
+            self.launch_type = launch_type.upper() if launch_type else "FARGATE"
```

Comment:
2. This still doesn't work because the only way to make launch_type = None is to first define the yaml file and then have "capacityProviderStrategy" defined in it.  In addition to the hassle of having a separate config file, we now have either a default launch_type or a default capacityProviderStrategy in place.  It makes run_config much more difficult to launch on specific resource type you want.

Reply:
The purpose of this PR is to allow the ECS Agent and ECSRun to use capacityProviderStrategy exclusively from the task_kwargs and should work.  This approach required the least amount changes to allow for this capability.  Driving this capability from config files is not as user friendly as say adding an additional argument for capacity provider in ECS Agent and the CLI. Why does this make launching to a specific resource type more difficult?

Reply:
Then you basically have to always specify "capacityProviderStrategy" in tasks_kwargs.  There are two problems.
1. In general we want the default to work.
2. I actually tried this and it's very difficult to get "capacityProviderStrategy" in tasks_kwargs working.  For example, in order to use FARGATE_SPOT, you'd have to set capacityProviderStrategy=[ { 'capacityProvider': 'FARGATE_SPOT', 'weight': 1, 'base': 0 }, ].  But deepcopy() can't handle this list properly.  There are also auto_scaling_group and other complicated capacityProviderStrategy I don't know if it would work or not.

In my practice, it's a lot easier to set the default strategy by aws console or cli, then just omit both fields in run_task api here.

Reply:
Right, the capacityProviderStrategy has to be passed in. This PR was tested against both ASGs and  Fargate and successfully deployed to both. Happy to provide examples of the ECSAgent YAML definition or the ECSRun run_task_kwargs used. 

Okay got it, so ECSAgent or ECSRun wouldn't set launch_type at all and the default capacityProviderStrategy of the cluster would be used. This would mean completely removing the defaults used for the launch_type logic in the agent and the CLI. Which I believe would require a lot of changes. Unfortunately that is outside the scope of what was planned for this PR but can be looked into shortly after this PR. Maybe a kwarg "default_capacity_provider" that drops both launch_type and capacityProvider if provided when set to True. 

The launch_type defaults exist because it's beneficial for new prefect users on an ECS infrastructure to quickly get started. It's doubtful that getting rid of these defaults is something the Prefect team would want to do.

Reply:
Agree that my suggests contradicts the current design.

Curious what's the correct way to pass capacityProviderStrategy in ECSRun under this PR ?  I tried
```python
ECSRun(run_task_kwargs=dict(capacityProviderStrategy=[ { 'capacityProvider': 'FARGATE_SPOT', 'weight': 1, 'base': 0 }, ], ...)
```
but `deepcopy()` didn't pass it through correctly.  I can see that YAML can work correctly but don't want to make this a default setting.

Reply:
Hmm  `dict(capacityProviderStrategy=[{'capacityProvider': 'FARGATE_SPOT', 'weight': 0, 'base': 1}]))`, should work. 

I just retested it and the event log for "RunTask" in Cloudtrail confirms that it works.  

![RunTask_cloudtrail](https://user-images.githubusercontent.com/34378029/156576851-86d3dec3-36ee-45d1-8b26-6b210ba63055.png)

Reply:
Can you go to ecs console and click the task to see if it's FARGATE_SPOT?  The default is FARGATE so even if the deepcopy fails it still works.  I observed the same thing.  Deepcopy somehow converted the value to a BoxList type.  Didn't investigate what happened.

Reply:
Yes the task seems to be using the FARGATE_SPOT capacityProvider.  

![image](https://user-images.githubusercontent.com/34378029/156617278-05ae3afd-ddae-4d2f-bb3c-dfdd2e8a9759.png)

Reply:
No intention to hijack the thread but in case this could save others' and my time, what I observe before and after merge_run_task_kwargs is:

![image](https://user-images.githubusercontent.com/93548332/156625875-0cfa8b3e-7fd9-47f1-b27e-f2126462d56f.png)

![image](https://user-images.githubusercontent.com/93548332/156626073-4e32b4ef-ad63-45df-9d88-4b4c941b1b0d.png)

Reply:
So this BoxList is because the server returns run_config as GraphQLResults that is a subclass of Box.  It doesn't seem to affect RunTask.  I think this PR is fine, although being able to use default strategy provider would be nice.  One hacky way to do that under this PR is to specify `capacityProviderStrategy=[]` in run_task_kwargs.